### DISCUSSION TITLE
Jekyll multilanguage websites

### DISCUSSION THREAD

**Discussion by EasyLifeBertolla** [community/community#39303] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

GitHub pages do not support any Jekyll plugins for i18n.

There are many nice plugins in development:
https://github.com/vwochnik/jekyll-language-plugin
https://github.com/Anthony-Gaudino/jekyll-multiple-languages-plugin
https://github.com/untra/polyglot 

None of these are whitelisted 😢
A Jekyll i18n support in GitHub pages would be highly appreciated by your international community.

Is there any plan to support these in the future, or will we be stuck forever?

**Comment by yoannchaudet** [comment]
Hello!

It is supported indirectly via https://github.blog/changelog/2022-07-27-github-pages-custom-github-actions-workflows-beta/ which is the direction the product is taking.

Steps:

- Enable "GitHub Actions" as the source of your Pages site
- Setup our "Jekyll" starter workflow ([source](https://github.com/actions/starter-workflows/blob/main/pages/jekyll.yml))
- Configure Jekyll the way you would do if you were not using Pages at all (`Gemfile`, `_config.yml`, etc.) - it is more work but you can use all the plugins in the world, as well as the latest version of Jekyll

**Reply by EasyLifeBertolla** [reply]
That's really nice. Thanks for the guidance, I will try this out.

**Reply by ProfaneServitor** [reply]
Github actions doesn't really support jekyll at the moment. See this issue: https://github.com/helaili/jekyll-action/issues/150

Is it possible to just whitelist `jekyll-polyglot` until that issue is fixed?

**Reply by yoannchaudet** [reply]
Do you have a link to your repo? There is nothing official in `helaili/jekyll-action`. It's a nice shortcut but if you want to leverage any plugins out there, you can just base your workflow of https://github.com/actions/starter-workflows/blob/main/pages/jekyll.yml.

We will not allow any more plugins in `pages-gem`.

**Reply by ProfaneServitor** [reply]
Here is an example of completely failing workflow:
https://github.com/ProfaneServitor/sdwui-docs/actions/runs/3775324578

It fails to install sass, but at least it's trying to compile the source.

Here is an example involving [said workflow](https://github.com/actions/starter-workflows/blob/main/pages/jekyll.yml):
https://github.com/ProfaneServitor/sdwui-docs/actions/runs/3782912683

It fails, because it can only serve precompiled pages.

**Reply by yoannchaudet** [reply]
![image](https://user-images.githubusercontent.com/14911070/209582790-8791629f-b875-423e-a599-0d2283ee60d6.png)

This only means you have an invalid dependencies mix. Have you try to run `bundler update` to fix that?

**Reply by ProfaneServitor** [reply]
Yes, I have. Also, it says above that it can't get any version of sass-embedded. 

```
NotImplementedError: sass_embedded for x86_64-linux-musl not available at
https://github.com/sass/dart-sass-embedded/releases/tag/1.57.1
```
